providers/apis/defi/raydium/types.go

```diff
@@ -0,0 +1,74 @@
+package raydium
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/gagliardetto/solana-go"
+)
+
+const (
+	// Name is the name of the Raydium API.
+	Name = "raydium_api"
+
+	// NormalizedTokenAmountExponent.
+	NormalizedTokenAmountExponent = 18
+)
+
+// TickerMetadata represents the metadata associated with a ticker's corresponding
+// raydium pool.
+type TickerMetadata struct {
+	// BaseTokenVault is the metadata associated with the base token's token vault
+	BaseTokenVault AMMTokenVaultMetadata `json:"base_token_vault"`
+
+	// QuoteTokenVault is the metadata associated with the quote token's token vault
+	QuoteTokenVault AMMTokenVaultMetadata `json:"quote_token_vault"`
+}
+
+// ValidateBasic checks that the solana token vault addresses are valid.
+func (metadata TickerMetadata) ValidateBasic() error {
+	if _, err := solana.PublicKeyFromBase58(metadata.BaseTokenVault.TokenVaultAddress); err != nil {
+		return err
+	}
+
+	if _, err := solana.PublicKeyFromBase58(metadata.QuoteTokenVault.TokenVaultAddress); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// AMMTokenVaultMetadata represents the metadata associated with a raydium AMM pool's
+// token vault. Specifically, we require the token vault address and the token decimals
+// for the token that the vault is associated with.
+type AMMTokenVaultMetadata struct {
+	// QuoteTokenAddress is the base58 encoded address of the serum token corresponding
+	// to this market's quote address
+	TokenVaultAddress string `json:"token_vault_address"`
+
+	// TokenDecimals is the number of decimals used for the token, we use this for
+	// normalizing the balance of tokens at the designated vault address
+	TokenDecimals uint64 `json:"token_decimals"`
+}
+
+// unmarshalMetadataJSON unmarshals the given metadata string into a TickerMetadata,
+// this method assumes that the metadata string is valid json, otherwise an error is returned.
+func unmarshalMetadataJSON(metadata string) (TickerMetadata, error) {
```

Comment:
Let's export and test

Reply:
This whole paradigm feels pretty fragile to me. I think we probably want to know how our arbitrary json objects are unmarhsalled way further up the stack and fail there instead of failing at the provider level. This feels like something that should be in the orchestrator--i.e. the request shouldn't even try to run unless we know we have the proper data to run it. I assume we'll want the same sort of behavior for any other provider using json metadata as well.

Reply:
Agree w/ @Eric-Warehime, we discussed offline, since the change to introduce this validation is a larger change / scoped outside of this PR, I'll be handling this in a follow-on